packages/sit-onyx/src/components/OnyxPagination/OnyxInlinePagination.vue

```diff
@@ -0,0 +1,207 @@
+<script lang="ts" setup>
+import { iconChevronLeftSmall, iconChevronRightSmall } from "@sit-onyx/icons";
+import { computed } from "vue";
+import { useDensity } from "../../composables/density.js";
+import {
+  SKELETON_INJECTED_SYMBOL,
+  useSkeletonContext,
+} from "../../composables/useSkeletonState.js";
+import { injectI18n } from "../../i18n/index.js";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import OnyxSkeleton from "../OnyxSkeleton/OnyxSkeleton.vue";
+import OnyxVisuallyHidden from "../OnyxVisuallyHidden/OnyxVisuallyHidden.vue";
+import type { OnyxPaginationProps } from "./types.js";
+
+const props = withDefaults(defineProps<OnyxPaginationProps>(), {
+  skeleton: SKELETON_INJECTED_SYMBOL,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the selected page changes.
+   */
+  "update:modelValue": [page: number];
+}>();
+
+const { t } = injectI18n();
+const { densityClass } = useDensity(props);
+const skeleton = useSkeletonContext(props);
+
+const hasReachedMin = computed(() => props.modelValue <= 1);
+const hasReachedMax = computed(() => props.modelValue >= props.pages);
+
+const displayPagesNumbers = computed<Array<number | string>>(() => {
+  const currentPage = props.modelValue;
+  const totalPages = props.pages;
+
+  const pagesToShow = new Set<number>();
+
+  pagesToShow.add(1);
+  pagesToShow.add(totalPages);
+
+  pagesToShow.add(currentPage);
+  pagesToShow.add(currentPage - 1);
+  pagesToShow.add(currentPage + 1);
+
+  if (currentPage === 1) pagesToShow.add(3);
+  if (currentPage === totalPages) pagesToShow.add(totalPages - 2);
+
+  const uniqueSortedPages = Array.from(pagesToShow)
+    .filter((page) => page >= 1 && page <= totalPages)
+    .sort((a, b) => a - b);
+
+  const finalPages: Array<number | string> = [];
+  let lastPageAdded: number | null = null;
+
+  for (const page of uniqueSortedPages) {
+    if (lastPageAdded !== null && page > lastPageAdded + 1) {
+      finalPages.push("ellipsis");
+    }
+    finalPages.push(page);
+    lastPageAdded = page;
+  }
+  return finalPages;
+});
+</script>
+
+<template>
+  <OnyxSkeleton v-if="skeleton" :class="['onyx-pagination-skeleton', 'onyx-text', densityClass]" />
+
+  <div
+    v-else
+    :class="[
+      'onyx-component',
+      'onyx-pagination',
+      'onyx-pagination__inline',
+      'onyx-text',
+      densityClass,
+    ]"
+    role="group"
+    :aria-label="t('pagination.label')"
+    :style="{ '--onyx-pagination-character-count': props.modelValue.toString().length }"
+  >
+    <button
+      class="onyx-pagination__inline__button"
+      :aria-label="t('pagination.previous')"
+      type="button"
+      :disabled="props.disabled || hasReachedMin"
+      @click="emit('update:modelValue', props.modelValue - 1)"
+    >
+      <OnyxIcon :icon="iconChevronLeftSmall" />
+    </button>
+    <template v-for="pageNumber in displayPagesNumbers" :key="`page-${pageNumber}`">
```

Comment:
```suggestion
    <template v-for="pageNumber in displayPagesNumbers" :key="pageNumber">
```

Is there a reason for the custom key?